web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -1,49 +1,116 @@
         var manga = document.createElement('div');
-        manga.innerText = bookmarks[i];
-        manga.onclick = function() {
-            clickedOnManga(this.innerText);
-        };
+        manga.innerText = bookmarks[i].label;
+        manga.onclick = () => this.onEntryClick(bookmarks[i].id);
         mangaList.appendChild(manga);
     }
 }
 
-function clickedOnManga(comicName) {
-    const engine = new BmcEngine();
-    engine.addEventListener(engine.events.register.error, err => {
-    });
-    engine.addEventListener(engine.events.register.complete, () => {
-        console.log('clickedOnManga!');
-        window.top.postMessage(`{"comicName": "${comicName.split('"').join('\\"')}"}`, '*');
-    });
-    engine.register(comicName, null, null);
+BmcMangaList.prototype.hideEntry = function(entry) {
+    entry.style.display = 'none';
 }
 
-function addEvents() {
-    var but = document.getElementById('hide-but');
-    but.onclick = function() {
-        showHideSidePanel(but);
+BmcMangaList.prototype.showEntry = function(entry) {
+    entry.style.display = '';
+}
+
+BmcMangaList.prototype.match = function(value, match) {
+    var lvalue = value.toLowerCase();
+    var lmatch = match.toLowerCase();
+    for (var i=0; i < lmatch.length; ++i) {
+        var idx = lvalue.indexOf(lmatch[i]);
+        if (idx === -1) {
+            return false;
+        }
+        lvalue = lvalue.slice(idx);
+    }
+    return true;
+}
+
+BmcMangaList.prototype.filter = function(filterStr) {
+    for (var i=0; i < this._node.childNodes.length; ++i) {
+        entry = this._node.childNodes[i];
+        if (this.match(entry.innerText, filterStr)) {
+            this.showEntry(entry);
+        } else {
+            this.hideEntry(entry);
+        }
     };
 }
 
-function showHideSidePanel(elem) {
-    var panel = document.getElementById("manga-list");
+
+function showHideSidePanel() {
+    var evData = {
+        type: "action",
+        action: null,
+    };
+    var btn = document.getElementById('hide-but');
+    var panel = document.getElementById("side-panel");
     if (panel.style.display === "none") {
+        evData.action = "ShowSidePanel",
         panel.style.display = '';
-        elem.innerText = '<';
-        elem.style.left = '';
-        elem.style.right = '0';
+        panel.style.width = 'calc(100vw - 16px)';
+        btn.innerText = '<';
+        btn.style.left = '';
+        btn.style.right = '0';
     } else {
+        evData.action = "HideSidePanel",
         panel.style.display = 'none';
-        elem.innerText = '>';
-        elem.style.left = '0';
-        elem.style.right = 'initial';
+        panel.style.width = '0';
+        btn.innerText = '>';
+        btn.style.left = '0';
+        btn.style.right = 'initial';
     }
+    // Notify top window of the SidePanel action
+    window.top.postMessage(evData, '*');
+}
+
+function addEvents(mangaList) {
+
+    // Clicking on the  >/< button will show/hide the panel
```

Comment:
Extra whitespace before `>/<`. Also, don't you think this might be easier to read:

```
// Clicking on the `>`/`<` buttons will show/hide the panel
```

?

Reply:
Will do ; wasnt really on my mine at time of writing.